calicoctl/commands/node/status.go

```diff
@@ -129,20 +131,40 @@ func printBGPPeers(ipv string) {
 		os.Exit(1)
 	}
 
-	buf := make([]byte, 1024)
+	// BIRD socket read timeout.
+	timeOut := 2 * time.Second
+	bufReader := bufio.NewReader(c)
 
-	n, err := c.Read(buf[:])
-	if err != nil {
-		fmt.Printf("Error: %v\n", err)
+	birdOut := ""
+
+	for {
+		// Set a time-out for reading from the socket connection.
+		// Read operation will fail if no data is received until the timeout.
+		c.SetReadDeadline(time.Now().Add(timeOut))
+
+		// Read string with \n as delim.
+		str, err := bufReader.ReadString('\n')
```

Comment:
Very inefficient looping here.  We are reading a string and then joining the strings together after they have been read.  And then below we split them by /n again and process each line separately.

Why not store a slice of strings and append each line to that slice.